cluster.go

```diff
@@ -1135,8 +1124,9 @@ func (c *Cluster) allocate(hash *cid.Cid, repl int) ([]peer.ID, error) {
 
 		// we don't have enough peers to pin
 		if len(candidateAllocs) < needed {
```

Comment:
Gotcha, thought it might have to do with keeping the 'allocate' strategy space open :) Yeah, hard to say whether it has much gain for an arbitrary allocation strategy/situation, but seems worth checking to catch potentially costly cases.